exaudfclient/base/javacontainer/javacontainer_impl.cc

```diff
@@ -49,16 +48,15 @@ JavaVMImpl::JavaVMImpl(bool checkOnly, bool noJNI,
     }
 }
 
-void JavaVMImpl::parseScriptOptions(std::unique_ptr<JavaScriptOptions::ScriptOptionsParser> scriptOptionsParser) {
-    JavaScriptOptions::Extractor extractor(*scriptOptionsParser);
+void JavaVMImpl::parseScriptOptions(std::unique_ptr<JavaScriptOptions::Extractor> extractor) {
 
-    DBG_FUNC_CALL(cerr,extractor.extract(m_scriptCode));
+    DBG_FUNC_CALL(cerr,extractor->extract(m_scriptCode));
 
     DBG_FUNC_CALL(cerr,setClasspath());
 
-    m_jvmOptions = std::move(extractor.moveJvmOptions());
+    m_jvmOptions = std::move(extractor->moveJvmOptions());
 
-    for (set<string>::iterator it = extractor.getJarPaths().begin(); it != extractor.getJarPaths().end();
+    for (set<string>::iterator it = extractor->getJarPaths().begin(); it != extractor->getJarPaths().end();
```

Comment:
This should save some redundant calls.
```suggestion
    const auto& jarPaths = extractor->getJarPaths();
    for (set<string>::iterator it = jarPaths.begin(); it != jarPaths.end();
```

Reply:
Changed it to 
```suggestion
        for (const auto & jar :  extractor->getJarPaths()) {
        addJarToClasspath(jar);
    }
```